crates/ra_analysis/src/imp.rs

```diff
@@ -305,6 +305,51 @@ impl AnalysisImpl {
             .collect()
     }
 
+    pub fn resolve_callable(&self, file_id: FileId, offset: TextUnit, token: &JobToken)
+        -> Option<(FnDescriptor, Option<usize>)> {
+
+        use ra_syntax::SyntaxKind;
+        use ra_syntax::ast::{ArgListOwner, Expr};
+        use ra_syntax::text_utils::{contains_offset_nonstrict};
+
+        let root = self.root(file_id);
+        let file = root.syntax(file_id);
+        let syntax = file.syntax();
+
+        // Find the calling expression and it's nameref
+        let call_expr: ast::CallExpr = find_node_at_offset(syntax, offset)?;
+
+        let name_ref = match call_expr.expr()? {
+            Expr::PathExpr(path_expr) => {
+                path_expr.path()?.segment()?.name_ref()?
+            },
+            _ => return None
+        };
+
+        // Resolve the function's NameRef
+        let file_symbols = self.index_resolve(name_ref, token);
```

Comment:
This doesn't always pick up the right function but it's probably good enough for now.